src/replication.c

```diff
@@ -4270,6 +4271,7 @@ void replicationResurrectProvisionalPrimary(void) {
      * drain. */
     replicationCreatePrimaryClientWithHandler(server.repl_transfer_s, server.repl_provisional_primary.dbid, NULL);
     memcpy(server.primary->replid, server.repl_provisional_primary.replid, CONFIG_RUN_ID_SIZE);
+    server.primary->replid[CONFIG_RUN_ID_SIZE] = '\0';
```

Comment:
I think we're protected here as well because replicationCreatePrimaryClientWithHandler will initialize it with a '\0' (although we set it with a wrong value initially, and then we have to fix it here, which is a bit weird).

Reply:
You're right, I missed that there is a copy of the null terminator happening inside the function above. So it should not have a bug.

I do think copying the extra byte reduces the cognitive load for this code, so I think we should proceed with the change.

I'll rename the PR to make it clear that this is a code cleanup.